### DISCUSSION TITLE
getNavigationBadge(): Return value must be of type ?string, int returned

### DISCUSSION THREAD

**Discussion by davideprevosto** [filamentphp/filament#12018] [Help]
### Package

Other

### Package Version

v3.2.60

### How can we help you?

Following this page: https://filamentphp.com/docs/3.x/panels/navigation, I added the following code in one of my Resources:

```php
public static function getNavigationBadge(): ?string
{
    return static::getModel()::count();
}
```

The IDE is notifying as per the attached screenshot:

<img width="961" alt="image" src="https://github.com/filamentphp/filament/assets/28837345/48cdac05-eb23-4f06-a58a-a78f5e3511ab">

...and I am also getting an error:

`App\Filament\Resources\ServerResource::getNavigationBadge(): Return value must be of type ?string, int returned`

What am I doing wrong?

Thank you

**Comment by leandrocfe** [comment]
```php
public static function getNavigationBadge(): ?string
{
    return (string) static::getModel()::count();
}
```
```php
public static function getNavigationBadge(): ?string
{
    return (string) YourModel::count();
}
```
?

**Reply by davideprevosto** [reply]
@leandrocfe yes of course it works adding (string).
The weird thing is: I copied from the guide, so I thought it was my bad instead of an error in the documentation.

**Reply by dissto** [reply]
Btw. I did not get the same error message when trying (tried on the sidebar navigation) the exact same code 🤔

**Reply by davideprevosto** [reply]
@dissto are you using the same package version? v3.2.60

**Reply by dissto** [reply]
v3.2.50

**Comment by mathieutu** [comment]
You probably have `declare(strict_types=1);` on top of your file.

If not, the `int` will be automatically converted to a string.
